src/chainstate/burn/db/sortdb.rs

```diff
@@ -2254,13 +2397,85 @@ impl <'a> SortitionHandleTx <'a> {
                     keys.push(db_keys::pox_last_anchor().to_string());
                     values.push("".to_string());
                 }
+                // if we've selected an anchor _and_ know of the anchor,
+                //  write the reward set information
+                if let Some(mut reward_set) = reward_info.known_selected_anchor_block_owned() {
+                    if reward_set.len() > 0 {
+                        // if we have a reward set, then we must also have produced a recipient
+                        //   info for this block
+                        let mut recipients_to_remove: Vec<_> = recipient_info.unwrap().recipients.iter().map(
+                            |(addr, ix)| (addr.clone(), *ix)).collect();
+                        recipients_to_remove.sort_unstable_by(|(_, a), (_, b)| b.cmp(a));
+                        // remove from the reward set any consumed addresses in this first reward block
+                        for (addr, ix) in recipients_to_remove.iter() {
+                            assert_eq!(&reward_set.remove(*ix as usize), addr,
+                                       "BUG: Attempted to remove used address from reward set, but failed to do so safely");
+                        }
+                    }
+
+                    keys.push(db_keys::pox_reward_set_size().to_string());
+                    values.push(db_keys::reward_set_size_to_string(reward_set.len()));
+                    for (ix, address) in reward_set.iter().enumerate() {
```

Comment:
Would it make more sense to combine the reward set entries into a single key/value pair?  I think you're going to have to load them all either way, so might as well minimize the number of MARF I/O ops.

Reply:
I don't think so. If they were a single key/value pair, the process of removing used entries would involve storing a copy of the whole list (minus removed entries) at every sortition, which is n^2/2 storage (which in this case would be a _lot_ of storage: 5000^2/2 * 21 bytes = 250mb per reward cycle).

By storing them individually, we can swap out the removed entries.